Rename CMake project to gz

chapulina opened the pull request:
* See https://github.com/ignition-tooling/release-tools/issues/718

scpeters commented:
may need to replace `ignition-` with `gz-` in the following script to fix some tests:

* https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/examples/test_core_child_requires_core_no_deps.bash

scpeters commented:
> may need to replace `ignition-` with `gz-` in the following script to fix some tests:
> 
> * https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/examples/test_core_child_requires_core_no_deps.bash

done in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/e54db0df90d24453a9a202798e7931be70ec45c0 and added an extra fix to the pkg-config file name in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/12be647e6e428af7af5eeee4573f21c7a40e01f5

scpeters commented:
I just noticed there is an `ignition-` that should change to `gz-` on the following line, but there's also a super weird `1.1` that makes no sense to me

* https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition-component.pc.in#L7

the `1.1` is also in [cmake/pkgconfig/ignition.pc.in](https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition.pc.in#L8)

scpeters commented on cmake/ignition-config.cmake.in:
```diff
@@ -58,9 +58,9 @@ set(@PKG_NAME@_CONFIG_INCLUDED TRUE)
+# Find gz-cmake, because we need its modules in order to find the rest of
 # our dependencies.
-find_dependency(ignition-cmake@IGNITION_CMAKE_VERSION_MAJOR@)
+find_dependency(@PROJECT_NAME@)
```
this is a mistake

scpeters replied:
fixed in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/e44d9dc0bfd2c715238247d6465a7447cb8b0089

scpeters replied:
fixed in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/e44d9dc0bfd2c715238247d6465a7447cb8b0089

scpeters commented:
> I just noticed there is an `ignition-` that should change to `gz-` on the following line, but there's also a super weird `1.1` that makes no sense to me
> 
> * https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition-component.pc.in#L7
> 
> the `1.1` is also in [cmake/pkgconfig/ignition.pc.in](https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition.pc.in#L8)

some explanation for `1.1` in https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-cmake/pull-requests/104/page/1

I think we can remove it

chapulina commented:
> regex replaces removing ignition to no-op

Good point, removed in [a477d3b](https://github.com/gazebosim/gz-cmake/pull/244/commits/a477d3b38a0b059fa3ba8e79ca914d4adb7a249d) since this PR will go along all downstream project name changes anyway.

chapulina commented:
> Good point, removed in [a477d3b](https://github.com/gazebosim/gz-cmake/pull/244/commits/a477d3b38a0b059fa3ba8e79ca914d4adb7a249d) since this PR will go along all downstream project name changes anyway.

Ah that broke some test cases. Reverted in [73ef90e](https://github.com/gazebosim/gz-cmake/pull/244/commits/73ef90eff1c88e0cb55dc13ba82484e2acc056fb). We can address that in follow-up PRs.
